mem.go

```diff
@@ -80,6 +81,21 @@ func (r RO) EqualString(s string) bool { return r.str() == s }
 // the same bytes.
 func (r RO) EqualBytes(b []byte) bool { return r.str() == string(b) }
 
+var builderPool = sync.Pool{
+	New: func() interface{} {
+		return new(strings.Builder)
+	},
+}
+
+// StringCopy returns m's contents in a newly allocated string.
+func (r RO) StringCopy() string {
+	buf := builderPool.Get().(*strings.Builder)
+	defer builderPool.Put(buf)
+	defer buf.Reset()
+	buf.WriteString(r.str())
```

Comment:
@josharian, can you please explain the reason for using `strings.Builder` instead of doing something like `string(r.unsafeBytes())`, wouldn't that be more efficient?

Reply:
What is `unsafeBytes()`? I don't see it anywhere.

Reply:
I was thinking about something like

```go
func (r RO) StringCopy() string {
        return string(unsafeBytes(r.str())
}

// can be wrong, just for illustration purposes
func unsafeBytes(s string) []byte {
	sh := (*reflect.StringHeader)(unsafe.Pointer(&s))

	var buf []byte
	bh := (*reflect.SliceHeader)(unsafe.Pointer(&buf))
	bh.Data = sh.Data
	bh.Cap = sh.Len
	bh.Len = sh.Len
	runtime.KeepAlive(&s) // prevent s from being freed before data is reachable via buf

	return buf
}
```

Reply:
Doing it this way keeps the usage of unsafe to a single unavoidable place in the code, which makes it easier to inspect for correctness.